contracts/proposers/BurnAuction.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.5.15;
+import { Chooser } from "./Chooser.sol";
+import { Bitmap } from "../libs/Bitmap.sol";
+import { Ownable } from "@openzeppelin/contracts/ownership/Ownable.sol";
+
+contract BurnAuction is Chooser, Ownable {
+    uint32 constant BLOCKS_PER_SLOT = 100;
+    uint32 constant DELTA_BLOCKS_INITIAL_SLOT = 1000;
+
+    // First block where the first slot begins
+    uint256 public genesisBlock;
+
+    // Bid is a structure indicating that
+    // "operator" is willing to burn certain
+    // "amount" of ether in order to forge a slot.
+    // "initialized" is used to indicate that the bid is valid (used to differenciate an empty bid)
+    struct Bid {
+        address payable operator;
+        uint128 amount;
+        bool initialized;
+    }
+
+    // auction is a relation of the best bid for each slot
+    mapping(uint32 => Bid) public auction;
+    // slot -> hasProposed
+    mapping(uint256 => uint256) public hasProposed;
+
+    /**
+     * @dev Event called when an operator beat the bestBid of the ongoing auction
+     */
+    event newBestBid(uint32 slot, address operator, uint128 amount);
+
+    /**
+     * @dev RollupBurnAuction constructor
+     * Set first block where the slot will begin
+     * Initializes auction for first slot
+     */
+    constructor() public {
+        genesisBlock = getBlockNumber() + DELTA_BLOCKS_INITIAL_SLOT;
+    }
+
+    /**
+     * @dev Receive a bid from an operator. If the bid is higher than the current bid it replace the existing bid
+     */
+    function bid() external payable {
+        uint32 auctionSlot = currentSlot() + 2;
+        require(
+            msg.value > auction[auctionSlot].amount,
+            "Your bid doesn't beat the current best"
+        );
+        // refund, check 0 case (it means no bids yet for the auction, so no refund)
+        // DISCUSSION: should we refund?
```

Comment:
Do you think we should remove the discussion comment here or leave it?

Reply:
Removing it sounds good